designdocs/batched_messages.md

```diff
@@ -10,98 +10,198 @@
+message and announce the batch location on-chain, instead of announcing each
+individual message.
+
+However, this idea does not go far enough. Batching allows us to support posting
+massive amounts of data, but it would still be expensive to post data of this
+size on chain.
+
+We can leverage off chain storage to make posting large message collections cheap, but we need to
+find a way to constrain the message types posted offchain so that message
+consumers know what types of data to expect. This document aims to explore what
+a system that does the above could look like.
 
 ## Goals and Non-Goals
-This specifies how batches are to be announced on chain; what is required and how a batch may be partially verified based on on-chain information.
+This specifies how messages are to be announced on chain; what is required and
+how a batch may be partially verified based on on-chain information.
+
+This document specifes how messages can be inferred from both schema format type
+and payload location.
 
-This document does not describe the types of DSNP messages that would be in batch announcements. Batch announcements must reference a schema through its ID number. A schema may or may not describe a DSNP message.
+This document also specifies how schemas will constrain the shape of off chain
+messages.
 
-This document does not discuss validation of batch file format located at the URI in the announcement, since
-the file format cannot be verified on-chain. For details about batch files themselves, see [DSNP Spec: Batch Publications](https://spec.dsnp.org/DSNP/BatchPublications).
+This document does not describe the types of DSNP Announcements that will be
+described by schemas. In theory, any message model can be supported.
+
+This document also does not discuss validation of either model or model type. If
+this type of validation is necessary, it should be described elsewhere.
 
 ## Proposal
 * All names are placeholders and may be changed.
 * Types may change as needed during implementation phase
 * Errors in the extrinsic(s) must have different, reasonably-named error enums for each type of error for ease of debugging.
 
-### Constants
-* `BatchPageSizeMax` - the maximum number of batch announcements that will be returned by the `get_batches` query
-* `BatchSizeMinBytes` - the minimum possible size of a valid batch file with 1 row
-
 ### Enums
-* `BatchFormat` - a list of supported formats for batch files. Currently only Parquet file format is supported, however,
-other formats are being considered.
+* `ModelType` - supported serialization formats for message payloads files. Currently only [Parquet](https://parquet.apache.org/docs/) and
+  [Avro](https://avro.apache.org/docs/current/) are supported.
+* `PayloadLocation` - The location of the payload. Can be either `OnChain` or `OffChain`.
+  * `OnChain`
+  * `OffChain`
+* `Payload`
+  * `OnChain`
+    * `source`: `MsaId`
+    * `payload`: `Vec<u8>`
+  * `IPFS`
+    * `payload_cid`: `TBD`
```

Comment:
1. Should likely be clear in that it is a v1 cid
2. We can use `Vec<u8>` here I think. That's what https://docs.rs/cid/0.8.5/cid/struct.CidGeneric.html#method.to_bytes returns.
```suggestion
    * `payload_cidv1`: `Vec<u8>`
```